src/middleware/verifyToken.ts

```diff
@@ -0,0 +1,32 @@
+import { getAuth } from "firebase-admin/auth";
+
+
+const verifyTokenMiddleware = async app => {
+
+  app.use(async (req, res, next) => {
+
+    let idToken
+
+    if (req.headers.authorization) {
+      idToken = req.headers.authorization
+    } else {
+      return res.status(403).json({ error: 'Unauthorized' })
+    }

```

Comment:
Não precisa desse if aqui, se o token for null vai cair no catch() do verifyIdToken() e já vai retornar o erro.

`const idToken = req.headers.authorization` é o suficiente